### PR TITLE
Add `source_id` to Authentications and Source has_many relationship

### PR DESCRIPTION
**Pull request by lindgrenj6**
https://issues.redhat.com/browse/RHCLOUD-10074

Adding a relationship between authentications and sources:

Source has_many Authentications

EDIT: also added `/sources/:id/authentications` endpoint per request from @rvsia

### CODE REVIEW

File: app/models/authentication.rb [hunk 1]
```diff
@@ -5,10 +5,19 @@ class Authentication < ApplicationRecord
...
 
   attribute :availability_status, :string
   validates :availability_status, :inclusion => { :in => %w[available unavailable] }, :allow_nil => true
+
+  before_validation :set_source
+
+  private
+
+  def set_source
+    self.source_id = resource.source_id
```
Comment (slemrmartin) [inline app/models/authentication.rb:21]: resource can be nil

File: db/migrate/20201120163548_add_source_to_authentications.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class AddSourceToAuthentications < ActiveRecord::Migration[5.2]
+  def up
+    add_column :authentications, :source_id, :bigint
+    Authentication.transaction do
+      Authentication.all.each do |auth|
```
Comment (slemrmartin) [inline db/migrate/20201120163548_add_source_to_authentications.rb:5]: Please use `find_each`

File: public/doc/openapi-3-v1.0.json [hunk 3]
```diff
@@ -1386,6 +1386,9 @@
             "example": "Endpoint",
             "type": "string"
           },
+          "source_id": {
```
Comment (slemrmartin) [inline public/doc/openapi-3-v1.0.json:1389]: should we update v1.0?

**Review by slemrmartin — CHANGES_REQUESTED** [review]
looks good, only few minor things

File: public/doc/openapi-3-v1.0.json (hunk 3 above)
Comment (lindgrenj6) [reply public/doc/openapi-3-v1.0.json:1435]: Not sure, I thought we were supposed to but should we just only update the latest?

File: app/models/authentication.rb (hunk 1 above)
Comment (lindgrenj6) [reply app/models/authentication.rb:21]: Well, the fun thing is that it won't allow it to be created unless the source exists, so `nil` won't work. I'll add `:optional => true` to the `belongs_to` statement to allow a blank source_id.

File: public/doc/openapi-3-v1.0.json (hunk 3 above)
Comment (slemrmartin) [reply public/doc/openapi-3-v1.0.json:1435]: I'm not sure too, theoretically yes, but practically v1.0 is used by someone so let's keep it updated

**Review by slemrmartin — APPROVED** [review]
:+1: LGTM, please generate sources-api-client for this PR

**Comment by lindgrenj6** [comment]
@slemrmartin https://github.com/RedHatInsights/sources-api-client-ruby/pull/24

**Pull request merged**